### ISSUE TITLE
`import/order` not working well with direct parent index

### ISSUE THREAD

**Issue by eTorAken** [benmosher/eslint-plugin-import#483]
Hello there,

I'm trying to set up a config for the `import/order` but I have issues when using `..` path to match with direct parent index (e.g. `../index.js`).

This is the config I use: 
```js
'import/order': [ // Enforce a convention in module import order.
  'warn',
  {
    'groups': [
      'builtin',
      ['external', 'internal'],
      ['parent', 'sibling', 'index'],
    ],
    'newlines-between': 'always',
  },
],
```

And here is an example of where I get a wrong warning:
```js
import moduleA from '../moduleA'; // import/order warning
import moduleB from '..';
```

It reports a warning on the first line: `import/order: There should be at least one empty line between import groups`. Thus, it is no more a problem when adding an ending `/` after module B's path:
```js
import moduleA from '../moduleA';
import moduleB from '../';
// everything is fine
```

According to me it's a bug, as I don't have to declare `import moduleA from '../moduleA/` to match an `index.js` file. What is your thoughts on this?

**Comment by benmosher** [comment]
Sounds like a bug, but we'll have to wait for @jfmengels to review.

**Comment by jfmengels** [comment]
Sounds like a bug to me in `importType` too. Currently on holidays, so won't be able to get to this before at least a week, so if someone wants to take a shot at this, go ahead :)
